app/src/androidTest/java/br/com/simplepass/loadingbutton/ProgressButtonStateIdlingResource.kt

```diff
@@ -0,0 +1,29 @@
+package br.com.simplepass.loadingbutton
+
+import androidx.test.espresso.IdlingResource
+import br.com.simplepass.loadingbutton.customViews.ProgressButton
+import br.com.simplepass.loadingbutton.presentation.State
+
+class ProgressButtonStateIdlingResource(private val progressButton: ProgressButton, private val expectedState: State = State.IDLE) : IdlingResource {
+
+    private var idle: Boolean = false
+    private var resourceCallback: IdlingResource.ResourceCallback? = null
+
+    override fun getName(): String = progressButton::class.java.simpleName
+
+    override fun isIdleNow(): Boolean {
+        idle = idle || progressButton.getState() == expectedState
+
+        if (idle) {
+            if (resourceCallback != null) {
```

Comment:
`resourceCallback?.onTransitionToIdle()`

Reply:
The check is made in the line above, so the `!!` is ok.